### PR TITLE
Add explanation for enabling libpodstats

### PR DESCRIPTION
**Pull request by pleimer**
Add a section explaining how to enable the gathering of service container metrics in the overcloud.

### CODE REVIEW

**Comment by pleimer** [comment]
I don't think my link to stf-connectors.yaml is working correctly at the moment

File: doc-Service-Telemetry-Framework/modules/con_service-resource-usage-metrics.adoc [hunk 1]
```diff
@@ -0,0 +1,28 @@
+
+// Module included in the following assemblies:
+//
+// <List assemblies here, each on a new line>
+
+// This module can be included from assemblies using the following include statement:
+// include::<path>/con_manifest-overrides.adoc[leveloffset=+1]
```
Comment (leifmadsen) [inline doc-Service-Telemetry-Framework/modules/con_service-resource-usage-metrics.adoc:7]:
```suggestion
// include::<path>/con_advanced-features.adoc [leveloffset=+1]
```

File: doc-Service-Telemetry-Framework/modules/con_service-resource-usage-metrics.adoc [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+// {context}, which ensures that the module has a unique ID even if it is
+// reused multiple times in a guide.
+//
+// In the title, include nouns that are used in the body text. This helps
+// readers and search engines find information quickly.
+// Do not start the title with a verb. See also _Wording of headings_
+// in _The IBM Style Guide_.
+[id="service-resource-usage-metrics_{context}"]
+= Service resource usage metrics
+Metrics pertaining to the resource usage of individual service contianers within an overcloud are not enabled by default. To gather these metrics, set `CollectdEnableLibpodstats: true` in the xref:configuring-red-hat-openstack-platform-overcloud-for-stf[stf-connectors.yaml] when deploying the overcloud.
```
Comment (leifmadsen) [inline doc-Service-Telemetry-Framework/modules/con_service-resource-usage-metrics.adoc:28]: So the `con` is for concept, so this file should be discussing the concept or the "what" a _Service resource usage metric_ is. If I was going to read this title I wouldn't really know what it meant or why I might look at it.

After the concept module, you'll create a `proc` or process module. The process module is for the actual work items you'd perform to enable this functionality. These process modules should be able to stand on their own, or link back to other process modules you might follow to get to this point.

The process will be fairly straight forward, but check out some of the other `proc` modules for a format of what you might do there. Process is probably just something like:

. Modify your `stf-connectors.yaml`. See [...] for more information about creating the `stf-connectors.yaml`.
. Add `CollectdEnableLibpodstats: true`
```
---
parameter_defaults:
  ...
  CollectdEnableLibpodstats: true
```

Or something like that. In the concept module I would also talk about any dependencies, and give an idea about what kinds of stats you might get, and why you might want this module enabled.

Comment (leifmadsen) [inline doc-Service-Telemetry-Framework/modules/con_service-resource-usage-metrics.adoc:28]:
```suggestion
Metrics pertaining to the resource usage of individual service containers within an overcloud are not enabled by default. To gather these metrics, set `CollectdEnableLibpodstats: true` in the xref:configuring-red-hat-openstack-platform-overcloud-for-stf[stf-connectors.yaml] when deploying the overcloud.
```

**Review by leifmadsen — DISMISSED** [review]

**Comment by leifmadsen** [comment]
You also need to include your module in doc-Service-Telemetry-Framework/assemblies/assembly_advanced-features.adoc

**Comment by leifmadsen** [comment]
```
diff --git a/doc-Service-Telemetry-Framework/assemblies/assembly_advanced-features.adoc b/doc-Service-Telemetry-Framework/assemblies/assembly_advanced-features.adoc
index 65c1486..6cd450c 100644
--- a/doc-Service-Telemetry-Framework/assemblies/assembly_advanced-features.adoc
+++ b/doc-Service-Telemetry-Framework/assemblies/assembly_advanced-features.adoc
@@ -69,3 +69,6 @@ include::../modules/proc_querying-metrics-data-from-multiple-clouds.adoc[levelof
 //Configuring ephemeral storage
 include::../modules/con_ephemeral-storage.adoc[leveloffset=+1]
 include::../modules/proc_configuring-ephemeral-storage.adoc[leveloffset=+2]
+
+//Service resource usage metrics
+include::../modules/con_service-resource-usage-metrics.adoc[leveloffset=+1]
```

File: doc-Service-Telemetry-Framework/modules/con_service-resource-usage-metrics.adoc (hunk 2 above)
Comment (pleimer) [reply doc-Service-Telemetry-Framework/modules/con_service-resource-usage-metrics.adoc:28]: Ah I see thanks

File: doc-Service-Telemetry-Framework/modules/con_service-resource-usage-metrics.adoc [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+// reused multiple times in a guide.
+//
+// In the title, include nouns that are used in the body text. This helps
+// readers and search engines find information quickly.
+// Do not start the title with a verb. See also _Wording of headings_
+// in _The IBM Style Guide_.
+[id="service-resource-usage-metrics_{context}"]
+= OpenStack service monitoring
+Conceptual overview:
+Monitoring the resource usage of the openstack services such as the APIs and other infrastructure processes can be very useful in identifying areas of the cloud that are under a lot of stress. This can be done by enabling the collectd-libpod-stats plugin to gather CPU and memory usage metrics for every container running in the overcloud.
```
Comment (JoanneOFlynn2018) [inline doc-Service-Telemetry-Framework/modules/con_service-resource-usage-metrics.adoc:29]: @pleimer This is a really good intro, thanks Paul. Knowing the why is super important. And thank you for including a prerequisite and procedure below. I will go through this later and apply RH docs style to it.

Comment (pleimer) [reply doc-Service-Telemetry-Framework/modules/con_service-resource-usage-metrics.adoc:29]: Thanks Joanne!

**Review by JoanneOFlynn2018 — APPROVED** [review]

**Comment by leifmadsen** [comment]
I'm going to get this merged down and imported in to the STF 1.1 documentation updates. If for some reason this misses 16.1.3 we'll need to comment it out downstream, but that's not a huge deal for upstream master docs.

File: doc-Service-Telemetry-Framework/modules/proc_monitoring-resource-usage-of-openstack-services.adoc [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+* You have created the `stf-connectors.yaml` file. For more information, see xref:configuring-red-hat-openstack-platform-overcloud-for-stf_completing-the-stf-configuration[].
+* You are using the most current version of {OpenStack}: {OpenStackVersion}.
+
+.Procedure
+
+. Open the `stf-connectors.yaml` file.
+
+. Add the following configuration to `parameter_defaults`:
++
+```
```
Comment (leifmadsen) [inline doc-Service-Telemetry-Framework/modules/proc_monitoring-resource-usage-of-openstack-services.adoc:42]: Does this actually load appropriately? This is markdown syntax :)

File: doc-Service-Telemetry-Framework/modules/proc_monitoring-resource-usage-of-openstack-services.adoc [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+  CollectdEnableLibpodstats: true
+```
```
Comment (leifmadsen) [inline doc-Service-Telemetry-Framework/modules/proc_monitoring-resource-usage-of-openstack-services.adoc:44]:
```suggestion
----
  CollectdEnableLibpodstats: true
----
```

**Review by leifmadsen — CHANGES_REQUESTED** [review]

**Comment by leifmadsen** [comment]
Did a local build and check, and this all loads fine. Merging!

**Review by leifmadsen — APPROVED** [review]

**Pull request merged**